macOS Sonoma unable to build postgres 16

**Describe the bug**
On macOS Sonoma 14.2.1, the attempt to install Postgres v16 through `asdf install postgres latest` fails, where the latest version is 16.1 as the date of this writing. It says icu library is missing:

```
  ...
  checking whether to build with ICU support... yes
  checking for icu-uc icu-i18n... no
  configure: error: ICU library not found
  If you have ICU already installed, see config.log for details on the
  failure.  It is possible the compiler isn't looking in the proper directory.
  Use --without-icu to disable ICU support.
  FAILED: install exited with an error
```

**Additional context**
One of the possible fixes is setting `PKG_CONFIG_PATH` as [this post](https://viggy28.dev/article/postgres-v16-icu-installation-issue/) suggests:

```
export PKG_CONFIG_PATH=/usr/local/opt/icu4c/lib/pkgconfig/
```

If you get icu4c through homebrew you can confirm the suggestion via `brew info icu4c`:

```
For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig"
```

Also, this error keeps the nightly build from passing since 2023-09-15, a day after [the release of Postgres v16](https://www.postgresql.org/about/news/postgresql-16-released-2715/), although I omit the CI logs as they evaporate in time.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,6 +18,6 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - name: asdf_plugin_test
-        uses: asdf-vm/actions/plugin-test@v1
+        uses: asdf-vm/actions/plugin-test@v3
         with:
           command: postgres --version

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -10,14 +10,21 @@ on:
     - cron: "0 0 * * *" # daily at midnight
 
 jobs:
-  plugin_test:
-    name: asdf plugin test
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+  plugin_test_ubuntu:
+    name: asdf plugin test (ubuntu)
+    runs-on: ubuntu-latest
     steps:
-      - name: asdf_plugin_test
+      - name: asdf_plugin_test_ubuntu
         uses: asdf-vm/actions/plugin-test@v3
+        with:
+          command: postgres --version
+  plugin_test_macos:
+    name: asdf plugin test (macos)
+    runs-on: macos-latest
+    steps:
+      - name: asdf_plugin_test_macos
+        uses: asdf-vm/actions/plugin-test@v3
+        env:
+          PKG_CONFIG_PATH: "/usr/local/opt/icu4c/lib/pkgconfig"
         with:
           command: postgres --version

